crm_claim_rma/models/crm_claim.py

```diff
@@ -102,61 +99,130 @@ def _onchange_invoice(self):
+                not group.users):
+            return False
+        return True
+
+    def _create_claim_lines(self):
+        self.ensure_one()
+        claim_lines = []
+        # here we should check if packages are enabled
+        if self.__check_packages_enabled():
+            # create claim lines from invoice lines with packages
+            claim_lines.extend(self.__create_lines_from_invoice_packages())
+        else:
+            # create claim lines from invoice lines
+            claim_lines.extend(self.__create_lines_from_invoice())
+
+        value = self._convert_to_cache(
+            {'claim_line_ids': claim_lines}, validate=False)
+        self.update(value)
+
+    def __create_lines_from_invoice(self):
+        claim_line = self.env['claim.line']
+        warehouse = self.warehouse_id
+        invoices_lines = self.invoice_id.invoice_line_ids.filtered(
+            lambda line: line.product_id.type in ('consu', 'product')
+        )
+        lines = []
+        for invoice_line in invoices_lines:
+            location_dest = claim_line.get_destination_location(
+                invoice_line.product_id, warehouse)
+            line = self._prepare_claim_line_from_invoice_line(
+                invoice_line,
+                location_dest
+            )
+            lines.append((0, 0, line))
+        return lines
+
+    def _prepare_claim_line_from_invoice_line(self, invoice_line,
+                                              location_dest):
+
+            line = {
+                'name': invoice_line.name,
+                'claim_origin': "none",
+                'invoice_line_id': invoice_line.id,
+                'product_id': invoice_line.product_id.id,
+                'product_returned_quantity': invoice_line.quantity,
+                'unit_sale_price': invoice_line.price_unit,
+                'location_dest_id': location_dest.id,
+                'state': 'draft',
+            }
+            line.update(invoice_line.invoice_id.warranty_values(
+                invoice_line.product_id,
+                self,
+            ))
+
+            return line
+
+    def __create_lines_from_invoice_packages(self):
+        claim_line = self.env['claim.line']
+        warehouse = self.warehouse_id
+        invoices_lines = self.invoice_id.invoice_line_ids.filtered(
+            lambda line: line.product_id.type in ('consu', 'product')
+        )
+        lines = []
+        for invoice_line in invoices_lines:
+            location_dest = claim_line.get_destination_location(
+                invoice_line.product_id, warehouse)
+            if not invoice_line.sale_line_ids:
+                continue
+            sale_line = invoice_line.sale_line_ids[0]
+            # here should be only one move
+            if not sale_line.procurement_ids:
+                continue
+            moves = sale_line.procurement_ids[0].move_ids.filtered(
+                lambda x: x.product_id == sale_line.product_id,
```

Comment:
it's a better practice to do:
```suggestion
                lambda x, p=sale_line.product_id: x.product_id == p,
```